activesupport/test/safe_buffer_test.rb

```diff
@@ -203,6 +203,30 @@ def test_titleize
     assert new_buffer.html_safe?, "should be safe"
   end
 
+  test "#split returns an Array of safe SafeBuffer instances when the original was safe" do
+    @buffer << "a"
+    @buffer << " "
+    @buffer << "b"
+
+    @buffer.split.each do |token|
+      assert_instance_of ActiveSupport::SafeBuffer, token
+      assert_predicate token, :html_safe?
+    end
+  end
+
+  test "#split returns an Array of unsafe SafeBuffer instances when the original was unsafe" do
+    @buffer << "a"
+    @buffer << " "
+    @buffer << "b"
+
+    @buffer.instance_variable_set :@html_safe, false
```

Comment:
@kamipo I've pushed up the test you mentioned. Is this the best way to set `@html_safe = false`, or is there a public method to call on the instance?

Reply:
We can use `gsub!` etc:

https://github.com/rails/rails/blob/ad12b996524244849222f16258e113b4dd3b19eb/activesupport/test/safe_buffer_test.rb#L112